lib/Core/Executor.cpp

```diff
@@ -1544,36 +1559,48 @@ void Executor::executeInstruction(ExecutionState &state, KInstruction *ki) {
 #else
           BasicBlock *caseSuccessor = si->getSuccessor(i);
 #endif
-          std::map<BasicBlock*, ref<Expr> >::iterator it =
-            targets.insert(std::make_pair(caseSuccessor,
-                           ConstantExpr::alloc(0, Expr::Bool))).first;
 
-          it->second = OrExpr::create(match, it->second);
+          // TODO add conditional
```

Comment:
@delcypher Good point, currently we `or`-combine different case values if they have the same target basic block. This increases the complexity of the generated constraint. If we handle them separate,  we can keep the simpler complexity but we will fork more often.
And yes, you convinced me, I'll update the comment. ;)